demos/mqtt/mqtt_demo_lightweight/mqtt_demo_lightweight.c

```diff
@@ -312,161 +291,69 @@ static uint16_t getNextPacketIdentifier()
 
 /*-----------------------------------------------------------*/
 
-static int connectToServer( const char * pServer,
-                            uint16_t port,
-                            int * pTcpSocket )
+static int connectToServerWithBackoffRetries( NetworkContext_t * pNetworkContext )
 {
-    int status = EXIT_SUCCESS;
-    struct addrinfo * pListHead = NULL, * pIndex, hint;
-    struct sockaddr * pServerInfo;
-    uint16_t netPort = htons( port );
-    socklen_t serverInfoLength;
-    struct timeval transportTimeout;
-
-    /* Set up hint structure, so that only TCP address structures are returned. */
-    memset( ( void * ) &hint, 0x00, sizeof( hint ) );
-    hint.ai_socktype = SOCK_STREAM;
-    /* Perform a DNS lookup on the given host name. */
-    status = getaddrinfo( pServer, NULL, &hint, &pListHead );
-
-    if( status != -1 )
+    int returnStatus = EXIT_SUCCESS;
+    bool backoffSuccess = true;
```

Comment:
Same as above - would it be better called `retryAttemptsRemaining`?

Reply:
I do agree that a better name is needed but I feel that `retryAttemptsRemaining` sounds like a quantity rather than a boolean. I am going to go with `retriesArePending` instead, but let me know if you disagree.